### PR TITLE
Factor artifact upload CI into custom GH action

### PR DESCRIPTION
**Pull request by nicholastmosher**
Adds a custom GH action for uploading artifacts from the `target/` directory. This effectively replaces a huge block of shell/action script with this:

```yaml
      - name: Upload Fluvio Artifacts
        uses: ./actions/upload-fluvio
        with:
          target: ${{ matrix.target }}
          artifact: |
            fluvio
            fluvio-run
            flv-test
```

This action will:

- Detect whether the workflow was executed in "release mode" i.e. Bors pushed it to staging
- Based on the release mode, select the artifacts from `target/debug` or `target/release`
- If given a compile `target`, it will select the artifacts from `target/${target}/` (e.g. `target/x86_64-unknown-linux-musl`)

### CODE REVIEW

**Comment by nicholastmosher** [comment]
Marking as Draft, just needed to open as PR in order to start testing in actual CI

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -188,45 +188,14 @@ jobs:
...
-        uses: actions/upload-artifact@v2
+      - name: Upload Fluvio Artifacts
+        uses: ./.github/actions/upload-fluvio
         with:
-          name: flv-test-${{ matrix.target }}
-          path: target/${{ matrix.target }}/release/flv-test
+          target: ${{ matrix.target }}
+          artifact: |
+            fluvio
+            fluvio-run
```
Comment (sehz) [inline .github/workflows/ci.yml:199]: `upload-artifact` allow multiple paths. So this could be done with that instead of customizing action?

Comment (nicholastmosher) [reply .github/workflows/ci.yml:199]: This action takes care of a bunch of variation that `upload-artifact` couldn't handle, such as:

- Are we in release mode or debug mode, and therefore do we upload from `target/debug` or `target/release`?
- Do we have a target other than host, so do we specify `target/debug` or `target/x86_64-unknown-linux-musl/debug`?

Basically this replaces this whole chunk from CI:

```
      # DEBUG MODE: Upload artifacts
      - name: Upload artifact - fluvio (DEBUG)
        if: ${{ !(github.event_name == 'push' && github.ref == 'refs/heads/staging') }}
        uses: actions/upload-artifact@v2
        with:
          name: fluvio-${{ matrix.target }}
          path: target/${{ matrix.target }}/debug/fluvio
      - name: Upload artifact - fluvio-run (DEBUG)
        if: ${{ !(github.event_name == 'push' && github.ref == 'refs/heads/staging') }}
        uses: actions/upload-artifact@v2
        with:
          name: fluvio-run-${{ matrix.target }}
          path: target/${{ matrix.target }}/debug/fluvio-run
      - name: Upload atrifact - flv-test (DEBUG)
        if: ${{ !(github.event_name == 'push' && github.ref == 'refs/heads/staging') }}
        uses: actions/upload-artifact@v2
        with:
          name: flv-test-${{ matrix.target }}
          path: target/${{ matrix.target }}/debug/flv-test

      # RELEASE MODE: Upload artifacts
      - name: Upload artifact - fluvio (RELEASE)
        if: github.event_name == 'push' && github.ref == 'refs/heads/staging'
        uses: actions/upload-artifact@v2
        with:
          name: fluvio-${{ matrix.target }}
          path: target/${{ matrix.target }}/release/fluvio
      - name: Upload artifact - fluvio-run (RELEASE)
        if: github.event_name == 'push' && github.ref == 'refs/heads/staging'
        uses: actions/upload-artifact@v2
        with:
          name: fluvio-run-${{ matrix.target }}
          path: target/${{ matrix.target }}/release/fluvio-run
      - name: Upload atrifact - flv-test (RELEASE)
        if: github.event_name == 'push' && github.ref == 'refs/heads/staging'
        uses: actions/upload-artifact@v2
        with:
          name: flv-test-${{ matrix.target }}
          path: target/${{ matrix.target }}/release/flv-test
```

**Comment by nicholastmosher** [comment]
Probably unneeded now

**Pull request closed without merging**